fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -1524,19 +1525,19 @@ private void transferToNonLeader(FrontendNodeType newType) {
             return;
         }
 
-        // transfer from INIT/UNKNOWN to OBSERVER/FOLLOWER
+        if (!isDefaultWarehouseCreated) {
```

Comment:
So isDefaultWarehouseCreated is useless if initDefaultWarehouse() is idempotent?

Reply:
yes

Reply:
will keep it for now, this PR will be focused on the fix of order of default warehouse initialization